### PR TITLE
Update rubocop and rubocop-performance

### PR DESCRIPTION
**Pull request by dorotheelaugwitz**
* Change `rubocop` version specification to `~> 1.8.1`
* Change `rubocop-performance` version specification to `~> 1.7.0`
* Update `.rubocop.yml` to match current shipcloud style guides
* Correct styling issues automatically with `rubocop -a`
* Fix remaining styling issues, mostly by adding magic comment
`# frozen_string_literal: true` to files

[ch7626](https://app.clubhouse.io/shipcloud/story/7626)

### CODE REVIEW

File: shipcloud.gemspec [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # coding: utf-8
+# frozen_string_literal: true
```
Comment (franziiiini) [inline shipcloud.gemspec:2]:
```suggestion
# frozen_string_literal: true

```
all the other files have an empty line following the magic comment, so we could do this here as well.

**Review by franziiiini — COMMENTED** [review]
One tiny suggestions, otherwise LGTM

**Comment by dorotheelaugwitz** [comment]
I added the most recent changes to our style guides @FranziGrieser.

**Review by franziiiini — APPROVED** [review]
thanks for putting on the rubocop hat and updating everything.

Ready to merge 🥳

**Pull request merged**